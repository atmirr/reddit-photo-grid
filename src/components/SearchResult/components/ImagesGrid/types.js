// @flow
export type ImagesGridProps = {
  posts: Array<Object>,
};
