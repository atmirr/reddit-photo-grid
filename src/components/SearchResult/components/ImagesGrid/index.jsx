// @flow
import React from 'react';
import type { Element } from 'react';
import { isEmpty } from 'lodash';
import styled from 'styled-components';
import ImageItem from './components/ImageItem';
import type { ImagesGridProps } from './types.js';

const Container = styled.div`
  max-width: 1400px;
  margin: 0 auto;
  padding: 30px 25px;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`;

function ImagesGrid({ posts }: ImagesGridProps): Element<'div'> {
  return (
    <Container>
      {!isEmpty(posts) &&
        posts.map(({ title, thumbnail }, index) => (
          <ImageItem title={title} thumbnail={thumbnail} key={index} />
        ))}
    </Container>
  );
}

export default ImagesGrid;
