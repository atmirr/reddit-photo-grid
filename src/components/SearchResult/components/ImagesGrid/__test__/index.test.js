import React from 'react';
import { render } from '@testing-library/react';
import ImagesGrid from '../index';

const posts = [
  {
    title: 'First sample post',
    thumbnail:
      'https://b.thumbs.redditmedia.com/lkIqoXa0Yhxwhy1rjgw0m6dJFnLVyPcZGpj0wzyTW9w.jpg',
  },
  {
    title: 'Second sample post',
    thumbnail:
      'https://a.thumbs.redditmedia.com/1RYKuLjrXVBam_FoGrBXfpdSxYMd52wU9J6RcZA0G88.jpg',
  },
];

describe('<ImagesGrid />', () => {
  it('should renders the component', () => {
    const { container } = render(<ImagesGrid />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should display images when posts prop is passed', () => {
    const { container } = render(<ImagesGrid posts={posts} />);
    expect(container.querySelectorAll('img').length).toEqual(2);
  });

  it("should contain post's title as the alt text for images", () => {
    const { queryByAltText } = render(<ImagesGrid posts={posts} />);
    expect(queryByAltText('First sample post')).toBeTruthy();
    expect(queryByAltText('Second sample post')).toBeTruthy();
  });
});
