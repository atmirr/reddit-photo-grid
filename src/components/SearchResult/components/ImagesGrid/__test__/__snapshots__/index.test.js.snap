// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImagesGrid /> should renders the component 1`] = `
<div
  class="sc-dlfnbm fvtGdp"
/>
`;
